### PR TITLE
Feature/occasion - ECOM-13 : Implement occasions screen

### PR DESCRIPTION
**Pull request by MuhamedUssama**
- Get occasions tab bar list from home screen.
- get request for all products end point.
- change products with the occasion index by its id.

Screenshot_1732619945

### CODE REVIEW

File: lib/features/home/presentation/tabs/occasions/occasions_screen.dart [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if (tabController.indexIsChanging) return;
+        viewModel.doIntent(
+          LoadProductsAction(occasions![tabController.index].id),
+        );
+      });
+
+      viewModel.doIntent(LoadProductsAction(occasions![0].id));
+    }
+  }
+
```
Comment (john-safwat) [inline lib/features/home/presentation/tabs/occasions/occasions_screen.dart:55]: tab controller and occasions list should be located in view model

File: lib/features/home/presentation/tabs/occasions/occasions_view_model/occasions_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import 'package:ecommerce_elevate/features/home/presentation/tabs/occasions/occasions_view_model/occasions_actions.dart';
+import 'package:ecommerce_elevate/features/home/presentation/tabs/occasions/occasions_view_model/occasions_states.dart';
+import 'package:injectable/injectable.dart';
+
+@injectable
+class OccasionsViewModel
+    extends BaseViewModel<OccasionsStates, OccasionsActions> {
+  final GetAllProductsListUseCase _allProductsUseCase;
+
+  @factoryMethod
```
Comment (john-safwat) [inline lib/features/home/presentation/tabs/occasions/occasions_view_model/occasions_view_model.dart:14]: no need for factory constractor here

File: lib/features/home/presentation/tabs/occasions/widgets/occasions_tabbar_widget.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      overlayColor: const WidgetStatePropertyAll<Color>(
+        Colors.transparent,
+      ),
+      tabAlignment: TabAlignment.start,
+      dividerHeight: 0,
+      indicatorColor: AppColors.pink,
+      isScrollable: true,
+      tabs: tabs,
+    );
+  }
```
Comment (john-safwat) [inline lib/features/home/presentation/tabs/occasions/widgets/occasions_tabbar_widget.dart:30]: all styling sholud be declerated in theme data and just use tab bar widget directly

File: lib/features/home/presentation/tabs/occasions/widgets/occasions_tabview_widget.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          return GridView.builder(
+            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+              crossAxisCount: 2,
+              childAspectRatio: 0.7,
+              crossAxisSpacing: 16,
+              mainAxisSpacing: 16,
+            ),
+            itemCount: state.products?.length ?? 0,
+            itemBuilder: (context, index) {
+              return ProductCardWidget(
```
Comment (john-safwat) [inline lib/features/home/presentation/tabs/occasions/widgets/occasions_tabview_widget.dart:30]: use skiltonizer here for loading effects

**Review by john-safwat — APPROVED** [review]
عاش جدا زي الف

**Review by yousefGamal966 — APPROVED** [review]
عاش اوي

**Review by mohamedAlaa-CS — APPROVED** [review]

**Pull request merged**